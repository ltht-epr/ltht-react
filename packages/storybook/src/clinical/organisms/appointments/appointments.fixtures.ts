import { Encounter, EncounterStatusCode, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock Source',
    },
  ],
  isRedacted: false,
  requestedWhen: '',
}

const redactedMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock Source',
    },
  ],
  isRedacted: true,
  requestedWhen: '',
}

const AppointmentOne: Encounter = {
  id: '346ca927-67e0-e951-a2c7-115087226fg2|appointment',
  isEnteredInError: false,
  metadata: mockMetadata,
  text: {
    text: 'See GP regarding sore throat',
    div: 'test',
  },
  type: [
    {
      text: 'See GP regarding sore throat',
      coding: [{ display: 'GP', code: '64706' }],
    },
  ],
  period: {
    start: {
      value: '2020-04-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  serviceProvider: {
    display: 'Beeston Village Surgery',
    typeName: 'GP',
  },
  status: EncounterStatusCode.Planned,
}

const AppointmentTwo: Encounter = {
  id: '941ca927-31e0-e911-a2c7-005087226fe4|appointment',
  isEnteredInError: false,
  metadata: mockMetadata,
  text: {
    text: 'Chest x-ray for recent car accident',
    div: 'test',
  },
  type: [
    {
      text: 'Chest x-ray for recent car accident',
      coding: [{ display: 'Hospital', code: '26706' }],
    },
  ],
  period: {
    start: {
      value: '2019-12-12T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  serviceProvider: {
    display: 'Chapel Allerton Hospital',
    typeName: 'Hospital',
  },
  status: EncounterStatusCode.Finished,
}

const AppointmentThree: Encounter = {
  id: '941ca927-34e0-e9113a2c7-005087226fe4|appointment',
  isEnteredInError: false,
  metadata: mockMetadata,
  text: {
    text: 'Chest x-ray for recent car accident',
    div: 'test',
  },
  type: [
    {
      text: 'Chest x-ray for recent car accident',
      coding: [{ display: 'Hospital', code: '26706' }],
    },
  ],
  period: {
    start: {
      value: '2019-12-12T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: EncounterStatusCode.Cancelled,
}

const AppointmentFour: Encounter = {
  id: '941ca927-34e0-e911-a2c7-005087226f44|appointment',
  isEnteredInError: false,
  metadata: redactedMetadata,
  text: {
    text: 'Chest x-ray for recent car accident',
    div: 'test',
  },
  type: [
    {
      text: 'Chest x-ray for recent car accident',
      coding: [{ display: 'Hospital', code: '26706' }],
    },
  ],
  period: {
    start: {
      value: '2019-12-12T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: EncounterStatusCode.Cancelled,
}

const appointments: Encounter[] = [AppointmentOne, AppointmentTwo, AppointmentThree, AppointmentFour]

export default appointments
