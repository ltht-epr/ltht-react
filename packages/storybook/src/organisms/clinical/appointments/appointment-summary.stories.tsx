import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import AppointmentSummary from '@ltht-react/appointment-summary'
import readme from '@ltht-react/appointment-summary/README.md'
import Card from '@ltht-react/card'
import encounters from './appointment-summary.fixtures'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const stories = storiesOf('Organisms - Clinical|Appointment', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Summary', () => (
  <Card>
    <Card.Header>
      <Card.Title>Appointments</Card.Title>
    </Card.Header>
    <Card.List>
      {encounters.map(encounter => (
        <Card.ListItem>
          <AppointmentSummary encounter={encounter} />
        </Card.ListItem>
      ))}
    </Card.List>
  </Card>
))
