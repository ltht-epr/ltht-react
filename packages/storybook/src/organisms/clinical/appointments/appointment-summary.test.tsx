import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import AppointmentSummary from '@ltht-react/appointment-summary'
import appointments from './appointment-summary.fixtures'

const Sut: React.FC = () => {
  return (
    <>
      {appointments.map(appointment => (
        <div key={appointment.id}>
          <AppointmentSummary encounter={appointment} />
        </div>
      ))}
    </>
  )
}

describe('AppointmentSummary', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Sut />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<Sut />)).toMatchSnapshot('wrapper mount')
  })
})
