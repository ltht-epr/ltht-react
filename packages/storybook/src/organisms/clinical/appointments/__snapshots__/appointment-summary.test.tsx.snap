// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentSummary matches snapshot: wrapper mount 1`] = `
.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: left;
}

.emotion-0 {
  color: #231f20;
}

.emotion-3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-2 {
  color: #231f20;
  text-align: left;
  margin-left: 0.5em;
}

.emotion-6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: right;
}

.emotion-4 {
  color: #768692;
  text-align: right;
}

.emotion-5 {
  color: #768692;
  text-align: right;
  font-size: smaller;
  padding-top: 0.25rem;
}

.emotion-25 {
  text-align: left;
}

.emotion-27 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: right;
}

.emotion-26 {
  color: #768692;
}

<Sut>
  <div
    key="346ca927-67e0-e951-a2c7-115087226fg2|appointment"
  >
    <AppointmentSummary
      encounter={
        Object {
          "id": "346ca927-67e0-e951-a2c7-115087226fg2|appointment",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock Source",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "period": Object {
            "start": Object {
              "kind": "DATE",
              "value": "2020-04-01T00:00:00+00:00",
            },
          },
          "serviceProvider": Object {
            "display": "Beeston Village Surgery",
            "typeName": "GP",
          },
          "status": "PLANNED",
          "text": Object {
            "div": "test",
            "text": "See GP regarding sore throat",
          },
          "type": Array [
            Object {
              "coding": Array [
                Object {
                  "code": "64706",
                  "display": "GP",
                },
              ],
              "text": "See GP regarding sore throat",
            },
          ],
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-7"
        >
          <Styled(div)>
            <div
              className="emotion-1"
            >
              <DateSummary
                datetime={
                  Object {
                    "kind": "DATE",
                    "value": "2020-04-01T00:00:00+00:00",
                  }
                }
              >
                <div
                  className="emotion-0"
                >
                  Apr-01,-2020
                </div>
              </DateSummary>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-3"
            >
              <AppointmentDescription
                encounter={
                  Object {
                    "id": "346ca927-67e0-e951-a2c7-115087226fg2|appointment",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock Source",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2020-04-01T00:00:00+00:00",
                      },
                    },
                    "serviceProvider": Object {
                      "display": "Beeston Village Surgery",
                      "typeName": "GP",
                    },
                    "status": "PLANNED",
                    "text": Object {
                      "div": "test",
                      "text": "See GP regarding sore throat",
                    },
                    "type": Array [
                      Object {
                        "coding": Array [
                          Object {
                            "code": "64706",
                            "display": "GP",
                          },
                        ],
                        "text": "See GP regarding sore throat",
                      },
                    ],
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  See GP regarding sore throat
                </div>
              </AppointmentDescription>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-6"
            >
              <AppointmentServiceProvider
                encounter={
                  Object {
                    "id": "346ca927-67e0-e951-a2c7-115087226fg2|appointment",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock Source",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2020-04-01T00:00:00+00:00",
                      },
                    },
                    "serviceProvider": Object {
                      "display": "Beeston Village Surgery",
                      "typeName": "GP",
                    },
                    "status": "PLANNED",
                    "text": Object {
                      "div": "test",
                      "text": "See GP regarding sore throat",
                    },
                    "type": Array [
                      Object {
                        "coding": Array [
                          Object {
                            "code": "64706",
                            "display": "GP",
                          },
                        ],
                        "text": "See GP regarding sore throat",
                      },
                    ],
                  }
                }
              >
                <div
                  className="emotion-4"
                >
                  Beeston Village Surgery
                </div>
              </AppointmentServiceProvider>
              <AppointmentStatus
                encounter={
                  Object {
                    "id": "346ca927-67e0-e951-a2c7-115087226fg2|appointment",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock Source",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2020-04-01T00:00:00+00:00",
                      },
                    },
                    "serviceProvider": Object {
                      "display": "Beeston Village Surgery",
                      "typeName": "GP",
                    },
                    "status": "PLANNED",
                    "text": Object {
                      "div": "test",
                      "text": "See GP regarding sore throat",
                    },
                    "type": Array [
                      Object {
                        "coding": Array [
                          Object {
                            "code": "64706",
                            "display": "GP",
                          },
                        ],
                        "text": "See GP regarding sore throat",
                      },
                    ],
                  }
                }
              >
                <div
                  className="emotion-5"
                >
                  Planned
                </div>
              </AppointmentStatus>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </AppointmentSummary>
  </div>
  <div
    key="941ca927-31e0-e911-a2c7-005087226fe4|appointment"
  >
    <AppointmentSummary
      encounter={
        Object {
          "id": "941ca927-31e0-e911-a2c7-005087226fe4|appointment",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock Source",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "period": Object {
            "start": Object {
              "kind": "DATE",
              "value": "2019-12-12T00:00:00+00:00",
            },
          },
          "serviceProvider": Object {
            "display": "Chapel Allerton Hospital",
            "typeName": "Hospital",
          },
          "status": "FINISHED",
          "text": Object {
            "div": "test",
            "text": "Chest x-ray for recent car accident",
          },
          "type": Array [
            Object {
              "coding": Array [
                Object {
                  "code": "26706",
                  "display": "Hospital",
                },
              ],
              "text": "Chest x-ray for recent car accident",
            },
          ],
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-7"
        >
          <Styled(div)>
            <div
              className="emotion-1"
            >
              <DateSummary
                datetime={
                  Object {
                    "kind": "DATE",
                    "value": "2019-12-12T00:00:00+00:00",
                  }
                }
              >
                <div
                  className="emotion-0"
                >
                  Dec-12,-2019
                </div>
              </DateSummary>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-3"
            >
              <AppointmentDescription
                encounter={
                  Object {
                    "id": "941ca927-31e0-e911-a2c7-005087226fe4|appointment",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock Source",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2019-12-12T00:00:00+00:00",
                      },
                    },
                    "serviceProvider": Object {
                      "display": "Chapel Allerton Hospital",
                      "typeName": "Hospital",
                    },
                    "status": "FINISHED",
                    "text": Object {
                      "div": "test",
                      "text": "Chest x-ray for recent car accident",
                    },
                    "type": Array [
                      Object {
                        "coding": Array [
                          Object {
                            "code": "26706",
                            "display": "Hospital",
                          },
                        ],
                        "text": "Chest x-ray for recent car accident",
                      },
                    ],
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  Chest x-ray for recent car accident
                </div>
              </AppointmentDescription>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-6"
            >
              <AppointmentServiceProvider
                encounter={
                  Object {
                    "id": "941ca927-31e0-e911-a2c7-005087226fe4|appointment",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock Source",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2019-12-12T00:00:00+00:00",
                      },
                    },
                    "serviceProvider": Object {
                      "display": "Chapel Allerton Hospital",
                      "typeName": "Hospital",
                    },
                    "status": "FINISHED",
                    "text": Object {
                      "div": "test",
                      "text": "Chest x-ray for recent car accident",
                    },
                    "type": Array [
                      Object {
                        "coding": Array [
                          Object {
                            "code": "26706",
                            "display": "Hospital",
                          },
                        ],
                        "text": "Chest x-ray for recent car accident",
                      },
                    ],
                  }
                }
              >
                <div
                  className="emotion-4"
                >
                  Chapel Allerton Hospital
                </div>
              </AppointmentServiceProvider>
              <AppointmentStatus
                encounter={
                  Object {
                    "id": "941ca927-31e0-e911-a2c7-005087226fe4|appointment",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock Source",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2019-12-12T00:00:00+00:00",
                      },
                    },
                    "serviceProvider": Object {
                      "display": "Chapel Allerton Hospital",
                      "typeName": "Hospital",
                    },
                    "status": "FINISHED",
                    "text": Object {
                      "div": "test",
                      "text": "Chest x-ray for recent car accident",
                    },
                    "type": Array [
                      Object {
                        "coding": Array [
                          Object {
                            "code": "26706",
                            "display": "Hospital",
                          },
                        ],
                        "text": "Chest x-ray for recent car accident",
                      },
                    ],
                  }
                }
              >
                <div
                  className="emotion-5"
                >
                  Finished
                </div>
              </AppointmentStatus>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </AppointmentSummary>
  </div>
  <div
    key="941ca927-34e0-e9113a2c7-005087226fe4|appointment"
  >
    <AppointmentSummary
      encounter={
        Object {
          "id": "941ca927-34e0-e9113a2c7-005087226fe4|appointment",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock Source",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "period": Object {
            "start": Object {
              "kind": "DATE",
              "value": "2019-12-12T00:00:00+00:00",
            },
          },
          "status": "CANCELLED",
          "text": Object {
            "div": "test",
            "text": "Chest x-ray for recent car accident",
          },
          "type": Array [
            Object {
              "coding": Array [
                Object {
                  "code": "26706",
                  "display": "Hospital",
                },
              ],
              "text": "Chest x-ray for recent car accident",
            },
          ],
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-7"
        >
          <Styled(div)>
            <div
              className="emotion-1"
            >
              <DateSummary
                datetime={
                  Object {
                    "kind": "DATE",
                    "value": "2019-12-12T00:00:00+00:00",
                  }
                }
              >
                <div
                  className="emotion-0"
                >
                  Dec-12,-2019
                </div>
              </DateSummary>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-3"
            >
              <AppointmentDescription
                encounter={
                  Object {
                    "id": "941ca927-34e0-e9113a2c7-005087226fe4|appointment",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock Source",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2019-12-12T00:00:00+00:00",
                      },
                    },
                    "status": "CANCELLED",
                    "text": Object {
                      "div": "test",
                      "text": "Chest x-ray for recent car accident",
                    },
                    "type": Array [
                      Object {
                        "coding": Array [
                          Object {
                            "code": "26706",
                            "display": "Hospital",
                          },
                        ],
                        "text": "Chest x-ray for recent car accident",
                      },
                    ],
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  Chest x-ray for recent car accident
                </div>
              </AppointmentDescription>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-6"
            >
              <AppointmentServiceProvider
                encounter={
                  Object {
                    "id": "941ca927-34e0-e9113a2c7-005087226fe4|appointment",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock Source",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2019-12-12T00:00:00+00:00",
                      },
                    },
                    "status": "CANCELLED",
                    "text": Object {
                      "div": "test",
                      "text": "Chest x-ray for recent car accident",
                    },
                    "type": Array [
                      Object {
                        "coding": Array [
                          Object {
                            "code": "26706",
                            "display": "Hospital",
                          },
                        ],
                        "text": "Chest x-ray for recent car accident",
                      },
                    ],
                  }
                }
              >
                <div
                  className="emotion-4"
                >
                  Mock Source
                </div>
              </AppointmentServiceProvider>
              <AppointmentStatus
                encounter={
                  Object {
                    "id": "941ca927-34e0-e9113a2c7-005087226fe4|appointment",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock Source",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2019-12-12T00:00:00+00:00",
                      },
                    },
                    "status": "CANCELLED",
                    "text": Object {
                      "div": "test",
                      "text": "Chest x-ray for recent car accident",
                    },
                    "type": Array [
                      Object {
                        "coding": Array [
                          Object {
                            "code": "26706",
                            "display": "Hospital",
                          },
                        ],
                        "text": "Chest x-ray for recent car accident",
                      },
                    ],
                  }
                }
              >
                <div
                  className="emotion-5"
                >
                  Cancelled
                </div>
              </AppointmentStatus>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </AppointmentSummary>
  </div>
  <div
    key="941ca927-34e0-e911-a2c7-005087226f44|appointment"
  >
    <AppointmentSummary
      encounter={
        Object {
          "id": "941ca927-34e0-e911-a2c7-005087226f44|appointment",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock Source",
              },
            ],
            "isRedacted": true,
            "requestedWhen": "",
          },
          "period": Object {
            "start": Object {
              "kind": "DATE",
              "value": "2019-12-12T00:00:00+00:00",
            },
          },
          "status": "CANCELLED",
          "text": Object {
            "div": "test",
            "text": "Chest x-ray for recent car accident",
          },
          "type": Array [
            Object {
              "coding": Array [
                Object {
                  "code": "26706",
                  "display": "Hospital",
                },
              ],
              "text": "Chest x-ray for recent car accident",
            },
          ],
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-7"
        >
          <AppointmentRedacted
            appointment={
              Object {
                "id": "941ca927-34e0-e911-a2c7-005087226f44|appointment",
                "metadata": Object {
                  "dataSources": Array [
                    Object {
                      "display": "Mock Source",
                    },
                  ],
                  "isRedacted": true,
                  "requestedWhen": "",
                },
                "period": Object {
                  "start": Object {
                    "kind": "DATE",
                    "value": "2019-12-12T00:00:00+00:00",
                  },
                },
                "status": "CANCELLED",
                "text": Object {
                  "div": "test",
                  "text": "Chest x-ray for recent car accident",
                },
                "type": Array [
                  Object {
                    "coding": Array [
                      Object {
                        "code": "26706",
                        "display": "Hospital",
                      },
                    ],
                    "text": "Chest x-ray for recent car accident",
                  },
                ],
              }
            }
          >
            <div
              className="emotion-25"
            >
              <DateSummary
                datetime={
                  Object {
                    "kind": "DATE",
                    "value": "2019-12-12T00:00:00+00:00",
                  }
                }
              >
                <div
                  className="emotion-0"
                >
                  Dec-12,-2019
                </div>
              </DateSummary>
            </div>
            <div
              className="emotion-27"
            >
              <RedactedDescription>
                <div
                  className="emotion-26"
                >
                  Insufficient privileges
                </div>
              </RedactedDescription>
            </div>
          </AppointmentRedacted>
        </div>
      </Styled(div)>
    </AppointmentSummary>
  </div>
</Sut>
`;
